flow-typed/mobx.js

```diff
@@ -127,12 +127,12 @@ export interface IObservableArray<T> extends Array<T> {
     clear(): T[];
     replace(newItems: T[]): T[];
     find(
-        predicate: (item: T, index: number, array: Array<T>) => boolean,
+        predicate: (item: T, index: number, array: Array<T>) => mixed,
         thisArg?: any,
         fromIndex?: number
     ): T | any;
     findIndex(
-        predicate: (item: T, index: number, array: Array<T>) => boolean,
+        predicate: (item: T, index: number, array: Array<T>) => mixed,
```

Comment:
The index is always numeric, right? Why not return number then?

Reply:
Again, `mixed` here is the return type of the `predicate`, not of `findIndex`. It's `mixed` and not `boolean` because anything that can be coerced to a boolean is okay as a return type.